### PR TITLE
docs(releasing): More release doc tweaks.

### PR DESCRIPTION
**Pull request by ttomsu**

### CODE REVIEW

File: RELEASING.adoc [hunk 1]
```diff
@@ -38,15 +38,15 @@ For GA releases this should be configured under the `central` Maven profile.
...
 ----
 
-. Ensure all Maven commands listed in `.kokoro/stage.sh` script can be executed successfully locally.
-You can add `-Dgpg.skip` in place of the actual GPG key locally.
+. Invoke the Kokoro `prod:cloud-java-frameworks/spring-cloud-gcp/stage` job.
+Note the staging repository ID in the logging output (i.e. `comgooglecloud-1234`).
 
-. Invoke the Kokoro `stage` job.
-Log in to OSS Sonatype and verify the contents of the staged package.
+. Log in to link:https://oss.sonatype.org/[OSS Sonatype] and verify the contents of the staged package. The URL should be similar to https://oss.sonatype.org/content/repositories/{STAGING_ID}/com/google/cloud/
```
Comment (meltsufin) [inline RELEASING.adoc:44]:
```suggestion
. Log in to link:https://oss.sonatype.org/[OSS Sonatype] and verify the contents of the staged package.
The URL should be similar to https://oss.sonatype.org/content/repositories/{STAGING_ID}/com/google/cloud/.
(If you don't have access to `com.google.cloud`, please make a request similar to https://issues.sonatype.org/browse/OSSRH-52371[this one] and ask for support from someone who already has access to this group ID.)
```

File: RELEASING.adoc [hunk 2]
```diff
@@ -38,15 +38,15 @@ For GA releases this should be configured under the `central` Maven profile.
...
 
-. Invoke to the Kokoko `promote` job.
+. Invoke to the Kokoko `prod:cloud-java-frameworks/spring-cloud-gcp/promote` job.
+Verify the artifacts have been published (https://search.maven.org/search?q=g:com.google.cloud%20AND%20a:spring-cloud-gcp, may take a few hours).
 
-. Make sure that the link:https://github.com/spring-io/start.spring.io/blob/master/start-site/src/main/resources/application.yml[Spring Boot Initializr] is updated to support the new version of Spring Cloud GCP.
+. Update start.spring.io by ensuring link:https://github.com/spring-io/start.spring.io/blob/master/start-site/src/main/resources/application.yml[Spring Boot Initializr] is updated to include the new version of Spring Cloud GCP.
 
 == Uploading Javadocs
```
Comment (meltsufin) [inline RELEASING.adoc:51]: Does this step really require a branch to have been created, if you're publishing from a release tag?
Speaking of the release tag, we need a step somewhere in the instructions for creating a release tag and updating the changelog.
Also, we need a step for incrementing the project version post-release.

**Review by meltsufin — CHANGES_REQUESTED** [review]
Can we also add a mention about how Snapshots are published??

File: RELEASING.adoc (hunk 2 above)
Comment (ttomsu) [reply RELEASING.adoc:53]: Branch? This is for any new version AFAICT. As in 2.0.0 --> 2.0.1 is a release that requires this step, but assuming the branch is `2.0.x` no new branch is needed.

As for branches - We briefly discussed following Spring's approach of using the minor version as the primary increment, but I don't think we committed to it. 

Your comments all point to the desire for a big checklist of steps to go through for each release. While I'm not opposed to that, I haven't ran through a whole release yet, so a fully fleshed out list would be mostly conjecture at this point.

Comment (meltsufin) [reply RELEASING.adoc:53]: We will need a branch for each of the Spring Cloud release trains.
I assume that we will be working on the Ilford release train for a while before jumping into the J____ release train sometime next year. So, we won't really need a `2.x` or `2.0.x` while `master` is aligned with Ilford.

Regarding release tags, we are currently missing them for RC1 and RC2. See: https://github.com/GoogleCloudPlatform/spring-cloud-gcp/releases. Also notice, Step 5 in [R2DBC repo](https://github.com/GoogleCloudPlatform/cloud-spanner-r2dbc/blob/master/RELEASING.adoc).

Comment (ttomsu) [reply RELEASING.adoc:53]: > we won't really need a 2.x or 2.0.x while master is aligned with Ilford.

So if I'm following your train of thought here, once Ilford is GA, the Spring branch will move on to `2021.0.0-SNAPSHOT` or whatever the J release train is. We will continue to use `2020.0.0` on our `master` until we upgrade to `2021.0.0-M1`, at which point we'll make a `2.0.x` branch and start emitting `2.1.0-SNAPSHOT` on `master`. Does that sound about right?

Comment (ttomsu) [reply RELEASING.adoc:53]: > we are currently missing them for RC1 and RC2

RC1 now has a tag, and once RC2 is out I'll tag it too.

Comment (meltsufin) [reply RELEASING.adoc:75]: > > we won't really need a 2.x or 2.0.x while master is aligned with Ilford.
> 
> So if I'm following your train of thought here, once Ilford is GA, the Spring branch will move on to `2021.0.0-SNAPSHOT` or whatever the J release train is. We will continue to use `2020.0.0` on our `master` until we upgrade to `2021.0.0-M1`, at which point we'll make a `2.0.x` branch and start emitting `2.1.0-SNAPSHOT` on `master`. Does that sound about right?

Exactly!

File: RELEASING.adoc [hunk 3]
```diff
@@ -38,15 +46,31 @@ For GA releases this should be configured under the `central` Maven profile.
...
-. Invoke the Kokoro `stage` job.
-Log in to OSS Sonatype and verify the contents of the staged package.
+. Verify that the new version has been published to Maven Central by checking https://repo.maven.apache.org/maven2/com/google/cloud/spring-cloud-gcp/[here].
+This might take a while.
 
-. Invoke to the Kokoko `promote` job.
+. https://github.com/GoogleCloudPlatform/spring-cloud-gcp/releases[Create] a new release on GitHub.
 
-. Make sure that the link:https://github.com/spring-io/start.spring.io/blob/master/start-site/src/main/resources/application.yml[Spring Boot Initializr] is updated to support the new version of Spring Cloud GCP.
+. Increment the project base _patch_ version. For example, from `2.0.0-SNAPSHOT` to `2.0.1-SNAPSHOT`.
```
Comment (meltsufin) [inline RELEASING.adoc:73]: Do we have a handy command for this? I remember you sharing it somewhere?

Comment (ttomsu) [reply RELEASING.adoc]: added

**Review by meltsufin — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**